crypto/ts/ts_rsp_sign.c

```diff
@@ -399,9 +399,9 @@ TS_RESP *TS_RESP_create_response(TS_RESP_CTX *ctx, BIO *req_bio)
                                                  TS_STATUS_REJECTION,
                                                  "Error during response "
                                                  "generation.") == 0) {
-                TS_RESP_free(ctx->response);
-                ctx->response = NULL;
             }
+            TS_RESP_free(ctx->response);
+            ctx->response = NULL;
```

Comment:
I am not sure this is quite right. IMO it was intentional to return a response even in case of failure. With your change the TS_RESP_CTX_set_status_info_cond() call is useless and also IMO the response status might indicate some other details about the failure.

Reply:
Ok.  So are you saying I should change `apps/ts.c` instead?  Is it is now, it uses the return value as error detection:

https://github.com/openssl/openssl/blob/f70863d9dddd3ce3420f0e07841475a7e9680ca9/apps/ts.c#L610-L611
...
https://github.com/openssl/openssl/blob/f70863d9dddd3ce3420f0e07841475a7e9680ca9/apps/ts.c#L638-L640

Reply:
The command exit code becomes zero, which means that gost-engine's tcl_tests don't notice that the command didn't perform right...

Reply:
Ah, looking further into gost-engine's tcl_tests, I see the consequence that you talk about...  [sigh] ok

Reply:
I've simply reverted this commit